monkestation/code/modules/bloodsuckers/monster_hunters/hunter_weapons.dm

```diff
@@ -223,6 +224,46 @@
 	enabled = FALSE
 	block_chance = 20
 
+/obj/item/melee/trick_weapon/beast_claw
+	name = "Beast Claw"
+	base_name = "Beast Claw"
+	desc = "The bones seem to still be twitching."
+	icon_state = "BoneClaw"
+	base_icon_state = "Claw"
+	w_class =  WEIGHT_CLASS_SMALL
+	block_chance = 20
+	base_force = 18
+	on_force = 23
+	throwforce = 10
+	wound_bonus = 25
+	bare_wound_bonus = 35
+	demolition_mod = 1.5 //ripping through doors and windows should be a little easier with a claw shouldnt it?
+	reach = 1
+	sharpness = SHARP_EDGED
+	hitsound = 'sound/weapons/fwoosh.ogg'
+	damtype = BURN
+	attack_verb_continuous = list("rips", "claws", "gashes", "tears", "lacerates", "dices", "cuts", "attacks")
+	attack_verb_simple = list("rip", "claw", "gash", "tear", "lacerate", "dice", "cut", "attack" )
+
+/obj/item/melee/trick_weapon/beast_claw/Initialize(mapload)
+	. = ..()
+	force = base_force
+	AddComponent(/datum/component/transforming, \
+		force_on = on_force, \
+		w_class_on = WEIGHT_CLASS_BULKY)
+	RegisterSignal(src, COMSIG_TRANSFORMING_ON_TRANSFORM, PROC_REF(on_transform))
+	RegisterSignal(src,WEAPON_UPGRADE, PROC_REF(upgrade_weapon))
+
+/obj/item/melee/trick_weapon/beast_claw/proc/on_transform/(obj/item/source, mob/user, active)
+	SIGNAL_HANDLER
+	balloon_alert(user, active ? "extended" : "collapsed")
```

Comment:
you're doing a lot of ternary checks in this proc. I'd recommend just doing an `if(active) do thing else do other thing for this entire code chunk

Reply:
i mostly just looked at the code from the other weapons and used it

Reply:
so i think ima leave it because it works when i tested it

Reply:
and if it aint broke